packages/dd-trace/src/llmobs/tagger.js

```diff
@@ -169,142 +169,156 @@ class LLMObsTagger {
           this._setTag(span, key, JSON.stringify(data))
         } catch {
           const type = key === INPUT_VALUE ? 'input' : 'output'
-          this._handleFailure(`Failed to parse ${type} value, must be JSON serializable.`, 'invalid_io_text')
+          this.#handleFailure(`Failed to parse ${type} value, must be JSON serializable.`, 'invalid_io_text')
         }
       }
     }
   }
 
-  _tagDocuments (span, data, key) {
-    if (data) {
-      if (!Array.isArray(data)) {
-        data = [data]
-      }
+  #tagDocuments (span, data, key) {
+    if (!data) {
+      return
+    }
 
-      const documents = data.map(document => {
-        if (typeof document === 'string') {
-          return { text: document }
-        }
+    if (!Array.isArray(data)) {
+      data = [data]
+    }
 
-        if (document == null || typeof document !== 'object') {
-          this._handleFailure('Documents must be a string, object, or list of objects.', 'invalid_embedding_io')
-          return undefined
-        }
+    const documents = []
+    for (const document of data) {
+      if (typeof document === 'string') {
+        documents.push({ text: document })
+        continue
+      }
 
-        const { text, name, id, score } = document
-        let validDocument = true
+      if (document == null || typeof document !== 'object') {
+        this.#handleFailure('Documents must be a string, object, or list of objects.', 'invalid_embedding_io')
+        continue
+      }
 
-        if (typeof text !== 'string') {
-          this._handleFailure('Document text must be a string.', 'invalid_embedding_io')
-          validDocument = false
-        }
+      const { text, name, id, score } = document
 
-        const documentObj = { text }
+      const valid = typeof text === 'string'
+      if (!valid) {
+        this.#handleFailure('Document text must be a string.', 'invalid_embedding_io')
```

Comment:
I'm not 100% sure about this as the calls to `#tagConditionalString` etc will throw in case the first argument isn't a string etc, so continuing here is technically changing behavior, right? But this just seems more correct without knowing more about this code

```suggestion
        this.#handleFailure('Document text must be a string.', 'invalid_embedding_io')
        continue
```